### PR TITLE
Mhv 44455 all triage team api

### PR DESCRIPTION
**Pull request by oleksii-morgun-va**
## Summary

- Adding `/alltriageteams` endpoint that returns a response with advanced attributes for `triage teams` by MHV SM Patient API
- Response from MHV SM API: 
`[
    {
        "associatedTriageGroups": 1,
        "associatedBlockedTriageGroups": 1
    },
    {
        "triageTeam": [
            {
                "triageTeamId": 1013155,
                "name": "***MEDICATION_AWARENESS_100% @ MOH_DAYT29",
                "stationNumber": "989",
                "blockedStatus": false,
                "preferredTeam": true,
                "relationshipType": "PATIENT"
            },
            {
                "triageTeamId": 2666935,
                "name": "Ademonow",
                "stationNumber": "989",
                "blockedStatus": true,
                "preferredTeam": true,
                "relationshipType": "PATIENT"
            },
        ]
    }
]`

- Response from `vets-api` `/my_health/v1/allrecipients`
`{
    "data": [
        {
            "id": "1013155",
            "type": "all_triage_teams",
            "attributes": {
                "triageTeamId": 1013155,
                "name": "***MEDICATION_AWARENESS_100% @ MOH_DAYT29",
                "stationNumber": "989",
                "blockedStatus": false,
                "preferredTeam": true,
                "relationshipType": "PATIENT"
            }
        },
        {
            "id": "2323849",
            "type": "all_triage_teams",
            "attributes": {
                "triageTeamId": 2323849,
                "name": "ALPHA_PCMM ASSOCIATION_DAYT29",
                "stationNumber": "989",
                "blockedStatus": false,
                "preferredTeam": false,
                "relationshipType": "PATIENT"
            }
        }
    ],
    "meta": {
        "associatedTriageGroups": 38,
        "associatedBlockedTriageGroups": 1,
        "sort": {
            "name": "ASC"
        }
    }
}`

## Related issue(s)
- https://jira.devops.va.gov/browse/MHV-44455 Update backend to allow for blocked triage group flag

## Testing done

- rspec testing included

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/my_health/app/controllers/my_health/v1/all_triage_teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module MyHealth
+  module V1
+    class AllTriageTeamsController < SMController
+      def index
+        resource = client.get_all_triage_teams(@current_user.uuid, use_cache?)
+        raise Common::Exceptions::InternalServerError if resource.blank?
```
Comment (mmoyer-va) [inline modules/my_health/app/controllers/my_health/v1/all_triage_teams_controller.rb:8]: We should eventually fix this line, which we have in several places in both SM and MR, per https://dsva.slack.com/archives/C018V2JCWRJ/p1694120211778619

**Review by mmoyer-va — APPROVED** [review]
Looks good! Left a comment but that can be fixed as part of a broader ticket if you want.

File: modules/my_health/app/controllers/my_health/v1/all_triage_teams_controller.rb (hunk 1 above)
Comment (oleksii-morgun-va) [reply modules/my_health/app/controllers/my_health/v1/all_triage_teams_controller.rb:8]: oh, that is right. I recall now. Yes, I think we can do an overall sweep in a separate PR. I would like to get your advice on what exactly needs to be done here 👍🏻

**Comment by oleksii-morgun-va** [comment]
@gia-lexa honestly I don't see a logical approach to split this up. This is addressing introducing new controller which is tied to a new route with respective response models and serializers, schemas - all are part of the same functionality.  Additionally, spec coverage + documentation updates, which to my understanding should go in hand. Open for any advice here.

File: app/models/all_triage_teams.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'common/models/base'
+
+# TriageTeam model
```
[the remark this answers is not in the document]
Comment (oleksii-morgun-va) [reply app/models/all_triage_teams.rb:5]: @gia-lexa thank you for catching that 🙀  It does indeed needs an update. Will push shortly

**Review by mmoyer-va — APPROVED** [review]

**Pull request merged**